src/client/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs

```diff
@@ -142,6 +142,40 @@ private void LogSuccessfulTelemetryToClient(AuthenticationResult authenticationR
                 telemetryEventDetails.SetProperty(TelemetryConstants.PopToken, authenticationResult.TokenType.Equals(Constants.PoPTokenType));
                 telemetryEventDetails.SetProperty(TelemetryConstants.RemainingLifetime, (authenticationResult.ExpiresOn - DateTime.Now).TotalMilliseconds);
                 telemetryEventDetails.SetProperty(TelemetryConstants.ActivityId, authenticationResult.CorrelationId);
+                telemetryEventDetails.SetProperty(TelemetryConstants.RefreshOn, 
+                    authenticationResult.AuthenticationResultMetadata.RefreshOn.HasValue ?
+                    DateTimeHelpers.DateTimeToUnixTimestampMilliseconds(authenticationResult.AuthenticationResultMetadata.RefreshOn.Value)
+                    : 0);
+
+                LogDatapointsToTelemetryClient(telemetryDatapoints, telemetryEventDetails);
+            }
+        }
+
+        private void LogDatapointsToTelemetryClient(Dictionary<string, string> telemetryDatapoints, MsalTelemetryEventDetails telemetryEventDetails)
+        {
+            if ((telemetryDatapoints == null) || (telemetryDatapoints.Count() == 0)) return;
+
+            string value;
+
+            if (telemetryDatapoints.TryGetValue(TelemetryCacheConstants.CacheUsed, out value))
```

Comment:
nit: 

```csharp
 if (telemetryDatapoints.TryGetValue(TelemetryCacheConstants.CacheUsed, out string value) && int.TryParse(value, out int cacheUsed)```

Reply:
i.e. you can declare the out variable inline, e.g. `out string x`

Reply:
Changed to class.